### PR TITLE
feat: Changes replication_configuration variable so it supports multiple rules

### PR DESCRIPTION
**Pull request by stefanwb**
Signed-off-by: Stefan Wessels Beljaars <swesselsbeljaars@schubergphilis.com>

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -201,17 +200,17 @@ resource "aws_s3_bucket_object_lock_configuration" "default" {
 }
 
 resource "aws_s3_bucket_replication_configuration" "default" {
-  for_each = local.replication_configuration
-  role     = var.replication_configuration.iam_role_arn
+  for_each = var.replication_configuration
```
[the remark this answers is not in the document]
Comment (stefanwb) [reply main.tf]: Thanks for the feedback. It's updated taking this into account

File: variables.tf [hunk 2]
```diff
@@ -101,10 +101,12 @@ variable "object_ownership_type" {
...
   type = object({
-    iam_role_arn       = string
-    dest_bucket        = string
-    dest_storage_class = string
-    rule_id            = string
+    iam_role_arn = string
+    rules = map(object({
+      dest_bucket        = string
+      dest_storage_class = string
+      rule_id            = string
```
Comment (marwinbaumannsbp) [inline variables.tf:108]:
```suggestion
      id            = string
```

File: main.tf [hunk 3]
```diff
@@ -205,13 +205,17 @@ resource "aws_s3_bucket_replication_configuration" "default" {
...
-    id     = var.replication_configuration.rule_id
-    status = "Enabled"
+  dynamic "rule" {
+    for_each = var.replication_configuration.rules
 
-    destination {
-      bucket        = var.replication_configuration.dest_bucket
-      storage_class = var.replication_configuration.dest_storage_class
+    content {
+      id     = rule.value["rule_id"]
```
Comment (marwinbaumannsbp) [inline main.tf:212]:
```suggestion
      id     = rule.value["id"]
```

File: variables.tf (hunk 2 above)
Comment (marwinbaumannsbp) [reply variables.tf:108]: its already part of the rules map

**Review by marwinbaumannsbp — APPROVED** [review]

**Pull request merged**